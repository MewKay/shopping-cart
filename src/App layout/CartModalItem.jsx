import PropTypes from "prop-types";
import styles from "./CartModalItem.module.css";

const CartModalItem = ({ image, title, price, quantity }) => {
  return (
    <li className={styles["item-card"]}>
      <div className={styles["image-container"]}>
        <img src={image} alt="" className={styles["item-image"]} />
      </div>
      <div className={styles["item-info"]}>
        <p className={styles["title"]}>{title}</p>
        <p className={styles["item-price"]}>
          {quantity} x ${price}
        </p>
      </div>
    </li>
  );
};

CartModalItem.propTypes = {
  image: PropTypes.string,
  title: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  quantity: PropTypes.number.isRequired,
};

export default CartModalItem;
