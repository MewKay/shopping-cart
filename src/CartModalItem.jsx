import PropTypes from "prop-types";

const CartModalItem = ({ image, title, price, quantity }) => {
  return (
    <li>
      <div>
        <img src={image} alt="" />
      </div>
      <div>
        <p>{title}</p>
        <p>
          {quantity} x ${price}
        </p>
      </div>
    </li>
  );
};

CartModalItem.propTypes = {
  image: PropTypes.string,
  title: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  quantity: PropTypes.number.isRequired,
};

export default CartModalItem;
