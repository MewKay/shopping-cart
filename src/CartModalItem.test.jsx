import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import CartModalItem from "./CartModalItem";

//Component props
const image = "https://image.placeholder.com/product";
const title = "Product 0";
const price = 49.5;
const quantity = 5;

describe("CartModalItem component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <CartModalItem
        image={image}
        title={title}
        price={price}
        quantity={quantity}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
