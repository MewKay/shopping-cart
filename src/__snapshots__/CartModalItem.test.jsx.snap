// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CartModalItem component > renders correctly 1`] = `
<div>
  <li>
    <div>
      <img
        alt=""
        src="https://image.placeholder.com/product"
      />
    </div>
    <div>
      <p>
        Product 0
      </p>
      <p>
        5
         x $
        49.5
      </p>
    </div>
  </li>
</div>
`;
